Extensions

iivchenko opened the pull request:
Little refactoring with extension methods.

Issue https://github.com/CrazyCouple/TActive/issues/9

Commit:
Add Extensions

T - ValidateNull
String - ValidateEmpty
Enumerable - ForEach
Collection - AddRange

Commit:
Turn of CA1062

CA can't see parameter validation
because it now implemented in extension methods.

Commit:
Refactor code with new Extensions

Commit:
fix line endings

mtarcha commented on Common/Common.Implementation/Extensions/StringExtensions.cs:
```diff
@@ -0,0 +1,31 @@
+        {
+            source.ValidateNull(name);
+
+            if (string.IsNullOrWhiteSpace(source))
```
For what you do 2 null validation here?
Don't you think that first validation "source.ValidateNull(name)" is excessive?

iivchenko replied:
You're right sweetheart =)
I just want to divide exceptions: 
- first if the string is NULL -> ArgumentNullException
- second if the string is "" or "   " (let it be an 'empty string') -> ArgumentException

so I used DRY

Why do I did that?
I think that nulll and empty string could be caused by different use cases. Null means that program logic fail to get a real value and empty string means it is a real value but the value is not appropriate.

mtarcha replied:
OK, you are right - but change the method name and add to xml doc. info about exceptions.

mtarcha replied:
And it is not always true - "Null means that program logic fail to get a real value" =)

Commit:
Improve StringExtensions

Rename ValidateEmpty => ValidateNullOrEmpty

Extend xml documentation of ValidateNullOrEmpty
with possible exceptions

mtarcha commented:
maladez =)
